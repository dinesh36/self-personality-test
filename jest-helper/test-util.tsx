import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import React from 'react';
import { render } from '@testing-library/react';
import { getStore } from '../src/store/store/store';

export const getComponentWithReduxWrapper = (cmp: React.ReactNode) => {
  const store = getStore();
  return renderer.create(
    <Provider store={store}>
      {cmp}
    </Provider>,
  );
};

export const getComponentWithReduxWrapperForReactTestingLib = (cmp: React.ReactNode) => {
  const store = getStore();
  return render(
    <Provider store={store}>
      {cmp}
    </Provider>,
  );
};
