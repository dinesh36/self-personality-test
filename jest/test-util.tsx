import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import React from 'react';
import { getStore } from '../src/store/store/store';

export const getComponentWithReduxWrapper = (cmp: React.ReactNode) => {
  const store = getStore();
  return renderer.create(
    <Provider store={store}>
      {cmp}
    </Provider>,
  );
};
