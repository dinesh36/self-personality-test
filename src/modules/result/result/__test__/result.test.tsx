import { getComponentWithReduxWrapper } from '@jestHelper/test-util';
import Result from '../result';

const mockSelector = jest.fn().mockReturnValue({ state: { testState: { result: '' } } });

jest.mock('react-redux', () => ({
  ...jest.requireActual('react-redux'),
  useSelector: () => mockSelector,
}));

describe('Result Component', () => {
  it('Should render the blank component when the result is not there', () => {
    const component = getComponentWithReduxWrapper(<Result />);
    const cmp = component.toJSON();

    expect(cmp).toMatchSnapshot();
  });

  it('should show the introvert result', () => {
    mockSelector.mockReturnValue({ state: { testState: { result: 'introvert' } } });
    const component = getComponentWithReduxWrapper(<Result />);
    const cmp = component.toJSON();

    expect(cmp).toMatchSnapshot();
  });
});
