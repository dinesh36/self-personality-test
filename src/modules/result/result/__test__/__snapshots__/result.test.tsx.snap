// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Result Component Should render the blank component when the result is not there 1`] = `
<div
  className="result"
>
  <div
    className="resultMediaContainer"
  >
    <img
      alt="extrovert"
      className="resultText"
      src="images/result/extrovert.png"
    />
    <div
      className="resultText"
    >
      <div
        className="subTitle"
      >
        You are more of an
         
        extrovert
      </div>
      Extroverts tend to be quite outgoing and talkative. They enjoy spending time with other people, and feel energized in social situations. Oftentimes, extroverts like being the focus of attention. As an extrovert, people probably describe you as friendly and outgoing.
    </div>
  </div>
  <button
    className="ant-btn css-dev-only-do-not-override-1n7nwfa ant-btn-primary button retakeTestButton"
    disabled={false}
    onClick={[Function]}
    type="button"
  >
    <span>
      Retake Test
    </span>
  </button>
</div>
`;

exports[`Result Component should show the introvert result 1`] = `
<div
  className="result"
>
  <div
    className="resultMediaContainer"
  >
    <img
      alt="extrovert"
      className="resultText"
      src="images/result/extrovert.png"
    />
    <div
      className="resultText"
    >
      <div
        className="subTitle"
      >
        You are more of an
         
        extrovert
      </div>
      Extroverts tend to be quite outgoing and talkative. They enjoy spending time with other people, and feel energized in social situations. Oftentimes, extroverts like being the focus of attention. As an extrovert, people probably describe you as friendly and outgoing.
    </div>
  </div>
  <button
    className="ant-btn css-dev-only-do-not-override-1n7nwfa ant-btn-primary button retakeTestButton"
    disabled={false}
    onClick={[Function]}
    type="button"
  >
    <span>
      Retake Test
    </span>
  </button>
</div>
`;
